ReactWindows/ReactNative.Net46/Modules/Storage/AsyncStorageModule.cs

```diff
@@ -274,6 +289,8 @@ public async void getAllKeys(ICallback callback)
 
         public override void OnReactInstanceDispose()
         {
+            _cancelSource.Cancel();
```

Comment:
Should we be taking the mutex here too?
What happens if a read/write is in progress where the mutex was taken.
We dispose the mutex and then the read/write completes and tries to release the mutex (which has now been disposed)

Another alternative I can think of is to replace all existing calls to _mutex.Release with calls have a new function SafeReleaseMutex which check IsCancelledOrDisposed before releasing the mutex.